stdlib/public/core/Reflection.swift

```diff
@@ -187,22 +188,38 @@ func _dumpWithMirror<TargetStream : OutputStreamType>(
   if let nam = name {
     print("\(nam): ", terminator: "", toStream: &targetStream)
   }
-  print(mirror.summary, terminator: "", toStream: &targetStream)
-
-  if let id = mirror.objectIdentifier {
-    if let previous = visitedItems[id] {
+  // This takes the place of the old mirror API's 'summary' property
+  print("\(String(reflecting: object))", terminator: "", toStream: &targetStream)
+
+  let id : ObjectIdentifier?
```

Comment:
Agreed, I've been thinking about the clearest way to break this function up. A revised version will be in the next update.